src/main/java/atas/model/session/AttributesList.java

```diff
@@ -0,0 +1,88 @@
+package atas.model.session;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import atas.commons.core.index.Index;
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+
+/**
+ * Wraps all data at the attribute-list level
+ */
+public class AttributesList implements ReadOnlyAttributesList {
+
+    protected final ObservableList<Attributes> attributes;
```

Comment:
Perhaps this could be private-access?

Reply:
Thanks for the suggestion, will be refactoring some methods to make this possible!